src/irust/cursor.rs

```diff
@@ -129,68 +144,84 @@ impl IRust {
         let x = if x.is_some() {
             x.unwrap()
         } else {
-            self.internal_cursor.screen_pos.0
+            self.pos.screen_pos.0
         };
 
         let y = if y.is_some() {
             y.unwrap()
         } else {
-            self.internal_cursor.screen_pos.1
+            self.pos.screen_pos.1
         };
 
-        self.cursor.goto(x as u16, y as u16)?;
+        self.internal_cursor.goto(x as u16, y as u16)?;
 
         Ok(())
     }
 
-    pub fn goto_cursor(&mut self) -> Result<(), IRustError> {
-        self.cursor.goto(
-            self.internal_cursor.screen_pos.0 as u16,
-            self.internal_cursor.screen_pos.1 as u16,
-        )?;
-        Ok(())
+    pub fn is_at_line_end(&self, irust: &IRust) -> bool {
+        irust.buffer.is_empty()
+            || self.pos.buffer_pos == StringTools::chars_count(&irust.buffer)
     }
 
-    pub fn at_line_end(&self) -> bool {
-        self.buffer.is_empty()
-            || self.internal_cursor.buffer_pos == StringTools::chars_count(&self.buffer)
+    pub fn screen_height_overflow_by_str(&self, irust: &IRust, out: &str) -> usize {
+        let new_lines =
+            (StringTools::chars_count(out) + self.pos.screen_pos.0) / irust.size.0;
+
+        self.screen_height_overflow_by_new_lines(irust, new_lines)
     }
 
-    pub fn move_cursor_left(&mut self, move_type: Move) -> Result<(), IRustError> {
-        self.internal_cursor.move_left(move_type);
-        self.goto_cursor()?;
+    pub fn screen_height_overflow_by_new_lines(&self, irust: &IRust, new_lines: usize) -> usize {
+        // if corrected y  + new lines < self.size.1 there is no overflow so unwrap to 0
+        (new_lines + self.pos.screen_pos.1).saturating_sub(irust.size.1)
+    }
+
+    pub fn save_cursor_position(&mut self) -> Result<(), IRustError> {
```

Comment:
I think we should just extract `save_position` function here, 2 functions that have almost the same name feel repetitive, also we should keep `save_position` name, exp:
```rust
fn save_position(&mut self) {
self.copy = //...
self.internal_cursor.save_position();
}
```